src/zope/interface/declarations.py

```diff
@@ -483,13 +500,30 @@ def classImplementsFirst(cls, iface):
 
 
 def _classImplements_ordered(spec, before=(), after=()):
+    # Elide everything already inherited.
+    # Except, if it is the root, and we don't already declare anything else
+    # that would imply it, allow the root through. (TODO: When we disallow non-strict
+    # IRO, this part of the check can be removed because it's not possible to re-declare
+    # like that.)
+    before = [
+        x
+        for x in before
+        if not spec.isOrExtends(x) or (x is Interface and not spec.declared)
+    ]
+    after = [
+        x
+        for x in after
+        if not spec.isOrExtends(x) or (x is Interface and not spec.declared)
+    ]
+
     # eliminate duplicates
     new_declared = []
     seen = set()
-    for b in before + spec.declared + after:
-        if b not in seen:
-            new_declared.append(b)
-            seen.add(b)
+    for l in before, spec.declared, after:
+        for b in l:
```

Comment:
Does pylint not complain about using lowercase-ell as a variable name?

Reply:
It probably would, but the check for valid names has to be disabled when working in zope repos because the false positives are overwhelming. (It's not the first one-letter temporary variable in this file.)

<img width="1193" alt="image" src="https://user-images.githubusercontent.com/1256082/78707733-9e285800-78d6-11ea-97f6-bf0494d54c1b.png">